### PR TITLE
Elide line:col info in "!!! related" errors when from default lib.d.ts files

### PR DESCRIPTION
**Pull request by jakebailey**
As noted on https://github.com/microsoft/TypeScript/pull/52317#discussion_r1081999615, this PR removes line:col information from the `!!! related` lines in our error baseline when the relation info comes from a default lib file.

Any time our libs are changed, loads of baselines change because position info differs; these "related" errors are the source of the bulk of them.

There are still a few baselines leftover that I'm not fixing in this PR; there are only 5 such examples in our repo, 4 of which are errors in tests with `@skipDefaultLibCheck: false`, the other being a tag in a completion baseline that links to a position in a lib file.

### CODE REVIEW

**Comment by jakebailey** [comment]
@sandersn I'm thinking this might the diff of #52328 smaller, if that's of interest.

File: src/compiler/program.ts [hunk 1]
```diff
@@ -744,16 +744,20 @@ function formatCodeSpan(file: SourceFile, start: number, length: number, indent:
...
     const relativeFileName = host ? convertToRelativePath(file.fileName, host.getCurrentDirectory(), fileName => host.getCanonicalFileName(fileName)) : file.fileName;
 
     let output = "";
     output += color(relativeFileName, ForegroundColorEscapeSequences.Cyan);
-    output += ":";
-    output += color(`${firstLine + 1}`, ForegroundColorEscapeSequences.Yellow);
-    output += ":";
-    output += color(`${firstLineChar + 1}`, ForegroundColorEscapeSequences.Yellow);
+
+    if (withPosition) {
```
Comment (jakebailey) [inline src/compiler/program.ts:753]: I noticed in #52328 that the symbol output elides these numbers already by replacing them with `--`; maybe that's a better idea than omitting it entirely?

**Comment by sandersn** [comment]
The error diffs on #52328 are already pretty small. But long-term this is a good idea.

File: src/compiler/program.ts (hunk 1 above)
Comment (sandersn) [reply src/compiler/program.ts:753]: 1. `--` is a better idea.
2. Adding a new overload is worth reflecting in the baselines IMO.

**Comment by jakebailey** [comment]
Yeah, I expected them to be huge and I thought my browser was just lagging when I clicked on it... nope, it was actually that short!

File: src/harness/harnessIO.ts [hunk 2]
```diff
@@ -596,7 +596,7 @@ export namespace Compiler {
                 .map(s => "!!! " + ts.diagnosticCategoryName(error) + " TS" + error.code + ": " + s);
             if (error.relatedInformation) {
                 for (const info of error.relatedInformation) {
-                    errLines.push(`!!! related TS${info.code}${info.file ? " " + ts.formatLocation(info.file, info.start!, formatDiagnsoticHost, ts.identity) : ""}: ${ts.flattenDiagnosticMessageText(info.messageText, IO.newLine())}`);
+                    errLines.push(`!!! related TS${info.code}${info.file ? " " + ts.formatLocation(info.file, info.start!, formatDiagnsoticHost, ts.identity, isDefaultLibraryFile(info.file.fileName)) : ""}: ${ts.flattenDiagnosticMessageText(info.messageText, IO.newLine())}`);
```
Comment (sheetalkamat) [inline src/harness/harnessIO.ts:599]: cant you instead do string replacement on all lines for errs to replace  `/.ts/lib.*.d.ts positions`

Comment (jakebailey) [reply src/harness/harnessIO.ts:599]: I can, it just seemed like it'd be a lot slower. The typeWriter already uses `/lib(.*)\.d\.ts/i` to detect this, but I was just unsure whether or not that was advisable if a test defined its own `lib.d.ts` files that weren't actually in `/.ts`.

Maybe I can just do:

```ts
if (isDefaultLibraryFile(info.file)) {
	location = location.replace(/(lib(?:.*)\.d\.ts):\d+:\d+:/i, "$1:")
}
```

If that's preferrable?

Comment (sheetalkamat) [reply src/harness/harnessIO.ts:599]: Yeah .. would prefer not having to add this to `program.ts`

Comment (jakebailey) [reply src/harness/harnessIO.ts]: Done.

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**